### ISSUE TITLE
Changing Vjoy Action settings conditions via the GUI is not recorded / registered / causes crashes (max recursion depth)

### ISSUE THREAD

**Issue by Lolo401** [muchimi/JoystickGremlinEx#27]
First, you're a legend for crunching through updates so quickly.

While checking update 13.40.8ex for the max recursion depth fix, the following 2 issues were noticed :

(1) A new file was created in JG 13.40.8Ex.
(2) Button 1 of a Virpil joystick was selected and remapped to Vjoy device 1 / Button 1.
(3) A container condition was added with a Vjoy condition on Vjoy Device1.
(4) Selecting a VJOY device was successful, however:

When trying to select the button and/or axis for the condition, JG Ex crashes. At the same time, the CPU usage rises to 82degrees / 100%.  After 2 minutes, the following message gets displayed :

![image](https://github.com/muchimi/JoystickGremlinEx/assets/157264378/16beb455-28d3-4306-bdb4-0f0b6522e944)
---------------------------------------------------------------------
(1) A new file was created as per before.
(2) A macro was created for button 2 with one action in the list.
(3) The action was changed to "Vjoy"in the action settings.

The option to select a button and/or a axis was missing :

![image](https://github.com/muchimi/JoystickGremlinEx/assets/157264378/bbde26a0-b0d1-4dd2-b2c5-ac0c01d5b85d)

**Comment by muchimi** [comment]
QT....   gotta love it.
Ok - let's see what it's up to now.  Probably the same issue as the drop down.  As you can tell, not a feature I use very much...

**Comment by Lolo401** [comment]
Yeah,  it is not breaking JG Ex strictu senso as I typically do my programming via VSCode directly in the xml.

I use the vjoy conditions a lot for the control of my Virpil LEDs.  I have a VJOY device used only for logic flags, not for control.  Ex: If Vjoy8/btn1 is pressed, set LED#18 of panel 2 becomes green, if released, set it to red etc.....  Therefore, I can control the same LED with multiple joystick buttons.  I am pretty sure there is a more elegant way to do this, but it works..

Cheers

**Comment by muchimi** [comment]
Please redownload https://github.com/muchimi/JoystickGremlinEx/releases/tag/13.40.8ex and see if that fixes the issue.
I was able to set a condition without a problem, but as I'm not entirely familiar with using conditions from code, probably something I'm missing.  There was another race condition handled.

**Comment by muchimi** [comment]
Also, interested in how you are controlling your VIRPIL leds - I don't have a throttle but this would be super helpful to add this feature to the code.

**Comment by Lolo401** [comment]
Issue is 95% fixed in the latest release :
(+) 
**No longer causes recursion depth (6-9 attempts, will keep on trying to replicate the issue).**

(-) 
**Sill no option to select buttons / axis if a VJOY device is selected as a source in a macro :**

JG Ex : 
![image](https://github.com/muchimi/JoystickGremlinEx/assets/157264378/cf02b46e-424d-4d92-8424-4ca393aa9e92)

JG 'Classic' 13.30:
![image](https://github.com/muchimi/JoystickGremlinEx/assets/157264378/b223d9bb-97ff-4ff1-ad14-b345f8098cbc)

**Changing a VJOY button status from Pressed to Released (in a condition or macro) does not get recorded and reverts back to pressed when the menu is exited**


https://github.com/muchimi/JoystickGremlinEx/assets/157264378/3ef7efc8-3f59-43e1-8768-ecc81078e964

Cheers

**Comment by Lolo401** [comment]
Damn, the last part of the video was incorrect, you need to leave the button menu :

https://github.com/muchimi/JoystickGremlinEx/assets/157264378/7fb77f63-4e14-4693-83cf-ce7525fee55a

**Comment by muchimi** [comment]
Can you try the pre-release?  It includes several fixes.  One of those was the controls would get reset by the UI and the fixes now block these updates so controls don't get re-initialized - so the settings would be lost.  I wonder if this will fix it.
